Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+                    value: enabled
+                });
+                break;
+        }
+    }
+
+
+    // This function keeps the app running even if you change domains.
+    var DOMAIN_DELAY = 100;
+    function onDomainChange(){
+        // turn it off on domain change
+        if (enabled) {
+            Script.setTimeout(function(){
+                startUpdate();
+                addAll();
+            }, DOMAIN_DELAY);
+        }
+    }
+
+
+    // This function simplifies the audio data from the Pal
+    var AVERAGING_RATIO = 0.05,
+        LOUDNESS_FLOOR = 11.0,
+        LOUDNESS_SCALE = 2.8 / 5.0,
+        LOG2 = Math.log(2.0),
+        AUDIO_PEAK_DECAY = 0.02;
+    function updateAudioLevel(uuid) {
+        var user = userStore[uuid];
+        if (!user) {
+            console.log("not a user");
+            return;
+        }
+
+        // scale audio
+        function scaleAudio(val) {
+            var audioLevel = 0.0;
+            if (val <= LOUDNESS_FLOOR) {
+                audioLevel = val / LOUDNESS_FLOOR * LOUDNESS_SCALE;
+            } else {
+                audioLevel = (val - (LOUDNESS_FLOOR - 1)) * LOUDNESS_SCALE;
+            }
+            if (audioLevel > 1.0) {
+                audioLevel = 1;
+            }
+            return audioLevel;
+        }
+
+        // the VU meter should work similarly to the one in AvatarInputs: log scale, exponentially averaged
+        // But of course it gets the data at a different rate, so we tweak the averaging ratio and frequency
+        // of updating (the latter for efficiency too).
+        var audioLevel = 0.0;
+        var avgAudioLevel = 0.0;
+
+        if (user) {
+            // we will do exponential moving average by taking some the last loudness and averaging
+            user.accumulatedLevel = AVERAGING_RATIO * (user.accumulatedLevel || 0) + 
+                (1 - AVERAGING_RATIO) * (user.audioLoudness);
+
+            // add 1 to insure we don't go log() and hit -infinity.  Math.log is
+            // natural log, so to get log base 2, just divide by ln(2).
+            audioLevel = scaleAudio(Math.log(user.accumulatedLevel + 1) / LOG2);
+
+            // decay avgAudioLevel
+            avgAudioLevel = Math.max((1 - AUDIO_PEAK_DECAY) * (user.avgAudioLevel || 0), audioLevel).toFixed(3);
+
+        }
+
+        userStore[uuid].audioLevel = audioLevel;
+        userStore[uuid].avgAudioLevel = avgAudioLevel;
+    }
+
+
+    // returns an array of avatarPaldata
+    function allAvatars() {
+        return AvatarManager.getPalData().data;
+    }
+    
+
+    // searches the user list to get the index of a user
+    function getIndexOfSettingsUser(uuid) {
+        if (settings.users.length) {
+            var index = settings.users.map(function (item) {
+                return item.uuid;
+            }).indexOf(uuid);
+            return index;
+        }
+        return -1;
+    }
+
+
+    // This function gets data to sort through
+    function sortData() {
```

Comment:
why is this necessary? what about this is sorting?

Reply:
It shouldn't be, Robin explained what it was for today, in the Loud app, it sorts the loudest users and displays only the 10 loudest.